test.disabled/ejabberd_tests/tests/push_pubsub_SUITE.erl

```diff
@@ -0,0 +1,415 @@
+                   ]);
+disable_stanza(JID, Node) ->
+    disable_stanza([
+                       {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+                       {<<"jid">>, JID},
+                       {<<"node">>, Node}
+                   ]).
+disable_stanza(JID) when is_binary(JID) ->
+    disable_stanza(JID, undefined);
+disable_stanza(Attrs) when is_list(Attrs) ->
+    escalus_stanza:iq(<<"set">>, [#xmlel{name = <<"disable">>, attrs = Attrs}]).
+
+enable_stanza(JID, Node) ->
+    enable_stanza(JID, Node, undefined).
+enable_stanza(JID, Node, FormFields) ->
+    enable_stanza(JID, Node, FormFields, ?NS_PUBSUB_PUB_OPTIONS).
+enable_stanza(JID, Node, FormFields, FormType) ->
+    escalus_stanza:iq(<<"set">>, [#xmlel{name = <<"enable">>, attrs = [
+        {<<"xmlns">>, <<"urn:xmpp:push:0">>},
+        {<<"jid">>, JID},
+        {<<"node">>, Node}
+    ], children = maybe_form(FormFields, FormType)}]).
+
+maybe_form(undefined, _FormType) ->
+    [];
+maybe_form(FormFields, FormType) ->
+    [make_form([{<<"FORM_TYPE">>, FormType} | FormFields])].
+
+make_form(Fields) ->
+    #xmlel{name = <<"x">>, attrs = [{<<"xmlns">>, ?NS_XDATA}, {<<"type">>, <<"submit">>}],
+           children = [make_form_field(Name, Value) || {Name, Value} <- Fields]}.
+
+make_form_field(Name, Value) ->
+    #xmlel{name = <<"field">>,
+           attrs = [{<<"var">>, Name}],
+           children = [#xmlel{name = <<"value">>, children = [#xmlcdata{content = Value}]}]}.
+
+
+
+%% ----------------------------------
+%% Other helpers
+%% ----------------------------------
+
+domain() ->
+    ct:get_config({hosts, mim, domain}).
+
+node_addr() ->
+    Domain = domain(),
+    <<"pubsub.", Domain/binary>>.
+
+rand_name(Prefix) ->
+    Suffix = base64:encode(crypto:rand_bytes(5)),
+    <<Prefix/binary, "_", Suffix/binary>>.
+
+pubsub_node_name() ->
+    rand_name(<<"princely_musings">>).
+
+pubsub_node() ->
+    {node_addr(), pubsub_node_name()}.
+
+parse_form(#xmlel{name = <<"x">>} = Form) ->
+    parse_form(exml_query:subelements(Form, <<"field">>));
+parse_form(Fields) when is_list(Fields) ->
+    lists:map(
+        fun(Field) ->
+            {exml_query:attr(Field, <<"var">>),
+             exml_query:path(Field, [{element, <<"value">>}, cdata])}
+        end, Fields).
+
+-spec rpc(M :: atom(), F :: atom(), A :: [term()]) -> term().
+rpc(M, F, A) ->
+    Node = ct:get_config({hosts, mim, node}),
+    Cookie = escalus_ct:get_config(ejabberd_cookie),
+    escalus_ct:rpc_call(Node, M, F, A, 10000, Cookie).
+
+bare_jid(JIDOrClient) ->
+    ShortJID = escalus_client:short_jid(JIDOrClient),
+    list_to_binary(string:to_lower(binary_to_list(ShortJID))).
+
+%% ----------------------------------------------
+%% REST mock handler
+setup_mock_rest() ->
+    Dispatch = cowboy_router:compile([
+		{'_', [
+			{"/[:level1/[:level2/[:level3/[:level4]]]]", ?MODULE, [{pid, self()}]}
+		]}
+	]),
+	{ok, _} = cowboy:start_http(http, 100, [{port, 8080}], [
```

Comment:
Port `8080` may often be used on local machines. How about allowing `cowboy` to choose a random port and using this port later?